### PR TITLE
Moving ECS modules around for a more functionality driven structure.

### PR DESCRIPTION
**Pull request by Rhuagh**
Also moved input rebinding to it's own crate as it can be used outside amethyst.

### CODE REVIEW

File: amethyst_input/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[package]
+name = "amethyst_input"
+version = "0.1.0"
+authors = ["Rhuagh <seamonr@gmail.com>"]
```
Comment (Xaeroxe) [inline amethyst_input/Cargo.toml]: I hate to be that guy, but given that I authored the majority of the functionality in this crate can you stick me as an additional author here?

Comment (Rhuagh) [reply amethyst_input/Cargo.toml]: Wow, did I forget that? Soooorrry :~(

**Comment by torkleyy** [comment]
Is there any need to merge this so fast? I planned to review this PR, but can't do that today anymore. I think we really have enough time to allow more reviews.

**Comment by nchashch** [comment]
@torkleyy Sure! So we will merge this after your review.

**Comment by Xaeroxe** [comment]
Examples still work with this commit so we're golden as far as I'm concerned.

**Pull request merged**